src/webui/abstractwebapplication.cpp

```diff
@@ -307,10 +308,12 @@ void AbstractWebApplication::increaseFailedAttempts()
 
 bool AbstractWebApplication::isAuthNeeded()
 {
-    return (env_.clientAddress != QHostAddress::LocalHost
-            && env_.clientAddress != QHostAddress::LocalHostIPv6
-            && env_.clientAddress != QHostAddress("::ffff:127.0.0.1"))
-            || Preferences::instance()->isWebUiLocalAuthEnabled();
+    qDebug("Checking auth rules against client address %s", qPrintable(env_.clientAddress.toString()));
+    if (Utils::Net::isLANAddress(env_.clientAddress))
```

Comment:
> As well as now it is unusable for others. For instance, your guesthouse gives IPs from sub-net 10.0.0.0/24. You set up a router in your room and assign to your laptop, phone and tablet IPs from 192.168.0.0/16. But you can't disable authentication for LAN clients, because all the guesthouse will be able to use your qBt instance.
 
This bypass feature is geared towards convenience. It's up to your network firewall to control what information can pass between subnets. Your host should also be running a firewall with additional rules. Anyone who needs additional control should turn towards one of these proven methods for enforcing network security.